test/e2e/addBook.spec.ts

```diff
@@ -0,0 +1,29 @@
+import { test, expect, Page } from "@playwright/test";
+import { LoginPage } from "../pages/LoginPage";
+import { BASE_URL, LOGIN_URL, VALID_EMAIL, VALID_PASSWORD } from "../helper-config";
+import { AddBookPage } from "../pages/AddBookPage";
+
+test.beforeEach(async ({ page }: { page: Page }) => {
+  await page.goto(LOGIN_URL);
+  const loginPage = new LoginPage({ page });
+  await loginPage.login(VALID_EMAIL, VALID_PASSWORD);
+});
+
+test("add a new book", async ({ page }: { page: Page }) => {
+  const TITLE: string = "One Piece4";
+  const PUBLISHER: string = "Luffy and Zoro";
+  const PUBLISHED: string = "1999";
+  const addBookPage = new AddBookPage({ page });
+  await addBookPage.navigateToAddBookPage();
+  await addBookPage.addBook(TITLE, PUBLISHER, PUBLISHED);
+  await expect(
+    page.getByText(
+      "Thank you for adding that book! Any more information you could provide would be wonderful!",
+      { exact: true }
+    )
+  ).toBeVisible();
+  await page.getByRole("button", { name: "Save" }).click();
```

Comment:
this should be part of the page-object. There should be no clicking happening in the specs file.

```suggestion
  await addBookPage.save();
```